x-pack/platform/plugins/shared/onechat/server/services/agents/modes/default/prompts.ts

```diff
@@ -138,34 +138,53 @@ export const getAnswerPrompt = ({
   customInstructions,
   discussion,
   handoverNote,
+  searchInterrupted = false,
   capabilities,
 }: {
   customInstructions?: string;
   discussion: BaseMessageLike[];
   handoverNote?: string;
+  searchInterrupted?: boolean;
   capabilities: ResolvedAgentCapabilities;
 }): BaseMessageLike[] => {
   const visEnabled = capabilities.visualizations;
+
+  let searchInterruptedMessages: BaseMessageLike[] = [];
+  if (searchInterrupted) {
+    searchInterruptedMessages = [
+      [
+        'ai',
+        'The research process was interrupted because it exceeded the maximum allowed steps. I cannot perform any more actions. Handing over for a final answer based on the information gathered so far',
+      ],
+      ['user', 'Ack. Proceed to answer as best as you can with the collected information'],
+    ];
+  }
+
   return [
     [
       'system',
       `You are an expert enterprise AI assistant from Elastic, the company behind Elasticsearch.
 
-Your role is to provide a clear, well-reasoned answer to the user's question using the information gathered by prior research steps.
-
-## INSTRUCTIONS
-- Carefully read the original discussion and the gathered information.
-- Synthesize an accurate response that directly answers the user's question.
-- Do not hedge. If the information is complete, provide a confident and final answer.
-- If there are still uncertainties or unresolved issues, acknowledge them clearly and state what is known and what is not.
+Your role is to be the **final answering agent** in a multi-agent flow. Your **ONLY** capability is to generate a natural language response to the user.
 
 ## IMPORTANT CONTEXT FROM THE PREVIOUS STEP
 The previous agent has completed its research and provided the following handover note:
 ---
-${handoverNote ?? 'No handover note was provided.'}
+${
+  handoverNote ?? searchInterrupted
+    ? 'Research was interrupted, please answer to the user as best as you can with the collected information'
+    : 'No handover note was provided.'
+}
```

Comment:
I might be wrong here, but if `handoverNote` is non-empty, will it return "Research was interrupted...", even when `searchInterrupted` is false?